arangod/IResearch/Search.cpp

```diff
@@ -480,7 +480,9 @@ ViewSnapshot::Links Search::getLinks() const {
   indexes.reserve(_indexes.size());
   for (auto const& [_, handles] : _indexes) {
     for (auto const& handle : handles) {
-      indexes.push_back(handle->lock());
+      if (auto index = handle->lock(); index) {
```

Comment:
Do we really need "; index" ?

Reply:
Yes! It's fix for single server. In single server then we get snapshot we use this method. If index already was deleted we shouldn't push it to snapshot (because acquiring snapshot will fail in this case)

Reply:
Ok. I bad understand your comment

Reply:
We can write it without ;index. But I don't like this style

Reply:
So here you want to emphasise that "index" is checked?

Reply:
it doesn't make much sense. 
I just prefer to write:
`if (auto p = ...; p)` instead of `if ((auto p = ...))`